plugins/rest/rest_auth.go

```diff
@@ -113,56 +121,142 @@ type tokenEndpointResponse struct {
 // oauth2ClientCredentialsAuthPlugin represents authentication via a bearer token in the HTTP Authorization header
 // obtained through the OAuth2 client credentials flow
 type oauth2ClientCredentialsAuthPlugin struct {
-	TokenURL     string    `json:"token_url"`
-	ClientID     string    `json:"client_id"`
-	ClientSecret string    `json:"client_secret"`
-	Scopes       *[]string `json:"scopes,omitempty"`
-
-	tokenCache    *oauth2Token
-	tlsSkipVerify bool
+	GrantType    string                 `json:"grant_type"`
+	TokenURL     string                 `json:"token_url"`
+	ClientID     string                 `json:"client_id"`
+	ClientSecret string                 `json:"client_secret"`
+	SigningKeyID string                 `json:"signing_key"`
+	Claims       map[string]interface{} `json:"additional_claims"`
+	IncludeJti   bool                   `json:"include_jti_claim"`
+	Scopes       []string               `json:"scopes,omitempty"`
+
+	signingKey       *keys.Config
+	signingKeyParsed interface{}
+	tokenCache       *oauth2Token
+	tlsSkipVerify    bool
 }
 
 type oauth2Token struct {
 	Token     string
 	ExpiresAt time.Time
 }
 
+func (ap *oauth2ClientCredentialsAuthPlugin) createAuthJWT(claims map[string]interface{}, signingKey interface{}) (*string, error) {
+	now := time.Now()
+	baseClaims := map[string]interface{}{
+		"iat": now.Unix(),
+		"exp": now.Add(10 * time.Minute).Unix(),
+	}
+	for k, v := range baseClaims {
+		claims[k] = v
```

Comment:
This throws `panic: assignment to entry in nil map` unless additionalClaims are set.

Reply:
Thanks, I'll fix and will add a test case.. though in fairness the JWT isn't really conforming to the targeted RFCs if entirely without claims.

Reply:
haha, agree :)

Reply:
Fixed in latest revision 👍